### PR TITLE
Adding TOA dev experience post.

### PR DESCRIPTION
**Pull request by AdamMc331**

### CODE REVIEW

File: _posts/2021-09-10-essential-dev-experience-concepts-android.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+* Detekt
+* Git Hooks
+
+The live stream link will have video markers to each of these sections, where you can see the implementation in more detail. For this blog post, we're just going to highlight what each of these concepts are, and why they were necessary for the [TOA application](https://github.com/adammc331/toa).
+
+## GitHub Actions
+
+GitHub Actions is a way to [automate workflows](https://github.com/features/actions) for your GitHub projects. The possibilities are endless - verifying unit tests pass, running lint checks, publishing to the play store, sharing internal builds, generating documentation - any process that you want to automate can be done via GitHub Actions. 
+
+You don't need to use Actions - there are many other CI tools out there such as Bitrise, Circle CI, Jenkins, and many more. We've opted to use GitHub Actions for this project because it is quick and easy to implement straight from GitHub, where this project is hosted anyways.
```
Comment (himattm) [inline _posts/2021-09-10-essential-dev-experience-concepts-android.md]: Redundant use of "many" here.

- Many other ci tools
- And many more

File: _posts/2021-09-10-essential-dev-experience-concepts-android.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+## Danger
+
+Danger is a tool for automating [common pull request chores](https://danger.systems/ruby). There are versions of Danger in multiple languages, but I prefer the Ruby gem. At first this may sound like the same thing as a GitHub Action, but it's a little more specific. Danger will run on Actions (or any other CI tool) to perform checks related to a Pull Request itself. An example of common Danger checks:
+
+* Does this PR have labels?
+* Does this PR have a description?
+* Did we assign a reviewer?
+* Is it too long? 
+* Did we modifile the CHANGELOG.md file?
```
Comment (himattm) [inline _posts/2021-09-10-essential-dev-experience-concepts-android.md]: Typo with modifile (unless this is a cool new term I don't know!)

File: _posts/2021-09-10-essential-dev-experience-concepts-android.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+For Android, I prefer to use [this Gradle plugin](https://github.com/jlleitschuh/ktlint-gradle) to set up Ktlint in my Android applications. For an example of how to quickly integrate the plugin, you can check out [this pull request](https://github.com/AdamMc331/TOA/pull/30/files). 
+
+## Detekt
+
+A similar tool that provides a lot of benefit is [Detekt](https://github.com/detekt/detekt). This is another static analysis tool but rather than just checking for formatting, it checks for other common code smells such as:
+
+* Magic numbers
+* Functions that are too long
+* Classic that have too many functions
```
Comment (himattm) [inline _posts/2021-09-10-essential-dev-experience-concepts-android.md]: Typo with classic

**Review by himattm — APPROVED** [review]
Just a few typos!

File: _posts/2021-09-10-essential-dev-experience-concepts-android.md (hunk 2 above)
Comment (AdamMc331) [reply _posts/2021-09-10-essential-dev-experience-concepts-android.md]: LMAO

**Pull request merged**